### PR TITLE
fix sealer apply repeatedly copying remote existing files

### LINKED ISSUE #164: Send sealer  binary file  repeatedly
#### What happened:
1,run sealer apply -f (my raw clusterfile,will apply infra)
2, shrink cluster node , i run sealer apply -f ($HOME/.sealer/Clusterfile,remote infra and cluster context already exist), it still send  sealer binary file . cost more time 

image

#### What you expected to happen:
no need to send sealer  binary file  repeatedly, because sealer  binary file is already exist
#### Environment:
- sealer version (use `sealer version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by touchuyht**
### Describe what this PR does / why we need it
Sealer apply will copy the files to remote host even if that file already exists because there is no checking process before copy. The new copy function will first check if the remote file exists and if it exists, it will check the md5sum of that remote file and compare it with local file's md5sum value. The copying process only happens when remote file does not exist or is stale.

### Does this pull request fix one issue?

Fixes #164

### CODE REVIEW

**Review by kakaZhou719 — APPROVED** [review]
LGTM

File: utils/ssh/scp.go [hunk 1]
```diff
@@ -217,8 +217,20 @@ func (s *SSH) copyLocalDirToRemote(host string, sshClient *ssh.Client, sftpClien
 	}
 }
 
+// check the remote file existence before copying
 // solve the sesion
 func (s *SSH) copyLocalFileToRemote(host string, sshClient *ssh.Client, sftpClient *sftp.Client, localPath, remotePath string) error {
+	var (
+		srcMd5, dstMd5 string
+	)
+	srcMd5 = FromLocal(localPath)
```
Comment (fanux) [inline utils/ssh/scp.go:226]: I think we can rename this function.

Comment (fanux) [reply utils/ssh/scp.go:226]: like GetLocalFileMD5

**Review by justadogistaken — APPROVED** [review]
LGTM

File: utils/ssh/scp.go (hunk 1 above)
Comment (touchuyht) [reply utils/ssh/scp.go]: solved

**Pull request merged**